pageserver/src/tenant/timeline.rs

```diff
@@ -3024,6 +3024,16 @@ impl Timeline {
 
         Some(HeatMapTimeline::new(self.timeline_id, layers))
     }
+
+    /// Returns true if the given lsn is or was an ancestor branchpoint.
+    pub(crate) fn is_ancestor_lsn(&self, lsn: Lsn) -> bool {
+        self.ancestor_lsn == lsn
+            || (self.ancestor_lsn == Lsn::INVALID
+                && self
+                    .remote_client
+                    .as_ref()
+                    .is_some_and(|rtc| rtc.is_previous_ancestor_lsn(lsn)))
```

Comment:
Future readers will be happy about a quick explainer here.

Also I don't know if the accessor `is_previous_ancestor_lsn()` is actually improving understandability of the code.

It would have been clearer to me if "lineage" were in the name, I wouldn't have had to jump to the definition and could have lived without my sggested comment.


```suggestion
            // If this timeline got detached from its ancestor timeline,
            // `self.ancestor_lsn` is copied into the `TimelineMetadataV2::lineage`
            // and then set to `Lsn::INVALID`.
            || (self.ancestor_lsn == Lsn::INVALID
                && self
                    .remote_client
                    .as_ref()
                    .is_some_and(|rtc| rtc.is_previous_ancestor_lsn(lsn)))
```

Reply:
Lineage is specifically not in the `TimelineMetadataV2` but in the `IndexPart`.

Oki, I'll monkey up some comment. I was thinking more of the avoiding locking the mutex with INVALID.